### ISSUE TITLE
Support OpenGL ES 2.0

### ISSUE THREAD

**Issue by hoffi** [fifengine/fifengine#793]
Would be great when FIFE supports OpenGL ES 2.0 so Unknown Horizons for example could run on a Raspberry Pi with better performance. 

Currently with SDL Rendering Engine the Game has about 2,5 FPS...

**Comment by helios2000** [comment]
Just to make it clear, if we want to go the way to OpenGL ES 2 and OpenGL 3 or above we have to change a lot. We've to change the render code from Fixed Function Pipeline to Programmable Pipeline. Propably the Renderers and all related features as Lights, Overlays and so on need a rewrite / review. We also have to consider what we want to do with the SDL backend and Fifechan.

We should start with https://github.com/fifengine/fifengine/issues/856 and https://github.com/fifengine/fifengine/issues/1014

**Comment by jakoch** [comment]
- User is requesting OpenGL ES 2.0 to improve performance on Raspberry Pi
- Currently there are 2 backends: 
   - SDL2
   - OpenGL 3.x (using GLEW, = desktop opengl, fixed functions)

TODO
- add rendering backend for OpenGL ES 2.0
  - src/engine/video/opengles/
- use GLES2 headers (GLES2/gl2.h)
- needs EGL for context creation on platforms like Raspberry Pi
- rendering code would need to be shader-based (not fixed-function)
- basically a conversion from immediate mode rendering to VBOs + shaders for lines, circles, images, tex
- SDL2 can be configured to use GLES2 
  - ref. SDL_GL_SetAttribute(SDL_GL_CONTEXT_PROFILE_MASK, SDL_GL_CONTEXT_PROFILE_ES)
- build system:
  - option(ENABLE_OPENGL_ES "Enable OpenGL_ES support" OFF)
  - find_package(OpenGLES QUIET)
